bin/src/lib.rs

```diff
@@ -15,22 +17,38 @@ static LOCK_NAME: &str = "iron-wallet";
 static LOCK_NAME: &str = "iron-wallet-dev";
 
 #[tokio::main]
-pub async fn run() -> color_eyre::Result<()> {
+pub async fn run() -> Result<()> {
     ethui_tracing::init()?;
     fix_path_env::fix()?;
 
     let args = ethui_args::parse();
-    let lock = NamedLock::create(LOCK_NAME)?;
 
-    let _guard = match lock.try_lock() {
-        Ok(g) => g,
-        Err(_) => {
-            ethui_broadcast::main_window_show().await;
-            return Ok(());
-        }
-    };
+    if let Some(command) = &args.command {
+        return handle_command(command, &args).await;
+    }
 
     app::EthUIApp::build(&args).await?.run();
 
     Ok(())
 }
+
+async fn handle_command(command: &Commands, args: &ethui_args::Args) -> Result<()> {
+    match command {
+        Commands::App { hidden: _ } => {
```

Comment:
```suggestion
        Commands::App {  } @ app_args => {
```

use these args as the argument instead of the global ones